### PR TITLE
Added Delete command

### PR DESCRIPTION
**Pull request by mrtwistor**
Added missing Delete command.

### CODE REVIEW

File: mathics/builtin/lists.py [hunk 1]
```diff
@@ -1683,6 +1683,94 @@ def callback(level):
 
         return Expression('List', *results)
 
+def del_part(expr,indices,evaluation):
+    def del_one(cur,pos):
```
Comment (suhr) [inline mathics/builtin/lists.py]: Is it really necessary to put a function inside a function?

Comment (mrtwistor) [reply mathics/builtin/lists.py]: I wouldn't say it's "necessary", no.  But these are just helper functions that are not intended to be called outside the function del_part, so they are local to that scope.

File: mathics/builtin/lists.py [hunk 2]
```diff
@@ -1684,6 +1684,96 @@ def callback(level):
...
+            return cur
+        elif pos == 0:
+            cur.head = Symbol('System`Sequence')
+            return cur
+        elif pos >= -l:
+            cur.leaves = cur.leaves[:l+pos] + cur.leaves[l+pos+1:]
+            return cur
+        else:
+            raise PartRangeError
+    def del_rec(self, cur, rest):
```
Comment (suhr) [inline mathics/builtin/lists.py]: Add a newline.

File: mathics/builtin/lists.py [hunk 3]
```diff
@@ -1684,6 +1684,96 @@ def callback(level):
...
+        if cur.is_atom():
+            raise PartDepthError
+        if len(rest)>1:
```
Comment (suhr) [inline mathics/builtin/lists.py]: Spaces around `>`.

File: mathics/builtin/lists.py (hunk 1 above)
Comment (suhr) [reply mathics/builtin/lists.py]: > Is that better?

I believe so.

File: mathics/builtin/lists.py (hunk 3 above)
Comment (mrtwistor) [reply mathics/builtin/lists.py]: ok

File: mathics/builtin/lists.py (hunk 2 above)
Comment (mrtwistor) [reply mathics/builtin/lists.py]: oops, I missed this.

File: mathics/builtin/lists.py [hunk 4]
```diff
@@ -1684,6 +1684,97 @@ def callback(level):
...
+                    part = self.del_rec(part,rest[1:])
+                    cur.leaves = cur.leaves[:l+pos] + [part] + cur.leaves[l+pos+1:]
+                    return cur
+                else:
+                    raise PartRangeError
+            except IndexError:
+                raise PartRangeError
+        else:
+            return self.del_one(cur, rest[0])
+    def del_part(self, expr,indices,evaluation):
```
Comment (suhr) [inline mathics/builtin/lists.py]: And here should be a newline too.

**Comment by suhr** [comment]
@wolfv

**Comment by wolfv** [comment]
Great! Fantastic contribution.

**Pull request merged**